core/src/main/java/org/elasticsearch/common/breaker/CircuitBreaker.java

```diff
@@ -32,6 +32,7 @@
     String FIELDDATA = "fielddata";
     String REQUEST = "request";
     String IN_FLIGHT_REQUESTS = "in_flight_requests";
+    String BUCKET = "bucket";
```

Comment:
I am still on the fence about having a dedicated breaker for it. When you look at this change, about 2/3 of the added lines are about the configuration of the breaker, only 1/3 are about memory accounting.

We need different breakers for eg. fielddata and requests because they have very different life spans, but bucket creation is just one of the many things that happen when executing requests. I don't think we should have such fine grained breakers for buckets, _source fields, query processing, etc. bringing each time 3 new settings and making the breaker tree more complex.